Informing readers about fallacies of distributed computing (#4455)
MicrosoftDocs/architecture-center: docs/patterns/category/design-implementation.md

Good design encompasses factors such as consistency and coherence in component design and deployment, maintainability to simplify administration and development, and reusability to allow components and subsystems to be used in other applications and in other scenarios. Decisions made during the design and implementation phase have a huge impact on the quality and the total cost of ownership of cloud hosted applications and services.

> [!NOTE]  
> In distributed systems:
> - Versioning isn't simple.
> - Observabality is not optional but mandatory.
> 

> 

Reviewer: Convert this text to be inline instead of using the NOTE syntax.  Find a way to stress the importance of these topics and relate it to the list below (presumably these will help solve those problems)

Good design encompasses factors such as consistency and coherence in component design and deployment, maintainability to simplify administration and development, and reusability to allow components and subsystems to be used in other applications and in other scenarios. Decisions made during the design and implementation phase have a huge impact on the quality and the total cost of ownership of cloud hosted applications and services.

Cloud workloads are prone to the [fallacies of distributed computing](https://wikipedia.org/wiki/Fallacies_of_distributed_computing). Some examples of cloud design fallacies are:

- The network is reliable
- Latency is zero
- Bandwidth is infinite
- The network is secure
- Topology doesn't change
- There is one administrator
- Component versioning is simple
- Observability implementation can be delayed

> Design patterns do not eliminate notions such as these but can help bring awareness, compenstations, and mitigations of them. Each cloud pattern has their own trade-offs, you need to pay attention more to why you are choosing a certain pattern rather than to how to implement it.

Reviewer 2: ```suggestion
Design patterns don't eliminate notions such as these but can help bring awareness, compensations, and mitigations of them. Each cloud pattern has its own trade-offs. You need to pay attention more to why you're choosing a certain pattern than to how to implement it.
```

Good design encompasses factors such as consistency and coherence in component design and deployment, maintainability to simplify administration and development, and reusability to allow components and subsystems to be used in other applications and in other scenarios. Decisions made during the design and implementation phase have a huge impact on the quality and the total cost of ownership of cloud hosted applications and services.

> These patterns are prone to [fallacies of distributed computing](https://en.wikipedia.org/wiki/Fallacies_of_distributed_computing) and even more: 
> - The network is reliable;
> - Latency is zero;
> - Bandwidth is infinite;
> - The network is secure;
> - Topology doesn't change;
> - There is one administrator;
> - Transport cost is zero;
> - The network is homogeneous;
> - Versioning is simple;
> - Observability is optional
> 
> Aforementioned assumptions are common but wrong. Before choosing any of the patterns, question your assumptions to ensure  you won’t fall for any of the above fallacies otherwise you will face consequences in terms of cost, unreliability, unavailability, lack of performance and / or security breaches. 
> 
> Remember that each of these patterns have their own trade-offs and you need to pay more attention to why you are choosing a certain pattern rather than to how to implement it. 
> 
> 

Reviewer: ```suggestion
Cloud workloads are prone to the [fallacies of distributed computing](https://wikipedia.org/wiki/Fallacies_of_distributed_computing). Some examples of cloud design fallacies are:

- The network is reliable
- Latency is zero
- Bandwidth is infinite
- The network is secure
- Topology doesn't change
- There is one administrator
- Component versioning is simple
- Observability implementation can be delayed

Design patterns do not eliminate notions such as these but can help bring awareness, compenstations, and mitigations of them. Each cloud pattern has their own trade-offs, you need to pay attention more to why you are choosing a certain pattern rather than to how to implement it.

```
